packages/@aws-cdk/aws-kinesisanalytics-flink/lib/types.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Available log levels for Flink applications.
+ */
+export enum LogLevel {
+  /** Debug level logging */
+  DEBUG = 'DEBUG',
+
+  /** Info level logging */
+  INFO = 'INFO',
+
+  /** Warn level logging */
+  WARN = 'WARN',
+
+  /** Error level logging */
+  ERROR = 'ERROR',
+}
+
+/**
+ * Granularity of metrics sent to CloudWatch.
+ */
+export enum MetricsLevel {
+  /** Application sends the least metrics to CloudWatch */
+  APPLICATION = 'APPLICATION',
+
+  /** Task includes task-level metrics sent to CloudWatch */
+  TASK = 'TASK',
+
+  /** Operator includes task-level and operator-level metrics sent to CloudWatch */
+  OPERATOR = 'OPERATOR',
+
+  /** Send all metrics including metrics per task thread */
+  PARALLELISM = 'PARALLELISM',
+}
+
+/**
+ * Interface for building AWS::KinesisAnalyticsV2::Application PropertyGroup
+ * configuration.
+ */
+export interface PropertyGroups {
+  readonly [propertyId: string]: {[mapKey: string]: string};
+}
+
+/**
+ * Available Flink runtimes for Kinesis Analytics.
+ *
+ * @example
+ * // Creating a new runtime that isn't in CDK yet.
+ * const runtime = new Runtime(FLINK-9_99);
+ */
+export class Runtime {
```

Comment:
Since we use static constants here, let's add an `public static of(value: string): Runtime` static factory method, and make the constructor private. It'll be easier for customers to find if they have to create a `Runtime` that the CDK doesn't support yet.